src/plugins/embeddable/common/lib/inject.ts

```diff
@@ -15,7 +15,7 @@ import { injectBaseEmbeddableInput } from './migrate_base_input';
 export const getInjectFunction = (embeddables: CommonEmbeddableStartContract) => {
   return (state: EmbeddableStateWithType, references: SavedObjectReference[]) => {
     const enhancements = state.enhancements || {};
-    const factory = embeddables.getEmbeddableFactory(state.type);
+    const factory = embeddables.getEmbeddableFactory?.(state.type);
```

Comment:
Looks like this function was marked optional in `CommonEmbeddableStartContract`, but that type isn't used in the client-side definition of `EmbeddableStart`. I understand this function was removed from the clientside version, but is there a reason it needed to be marked optional in the serverside one?

Reply:
Both client and server use `inject` so I typed `getEmbeddableFactory` as optional since its not provided by the client.